extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/FilterResultUpdater.java

```diff
@@ -126,11 +129,19 @@ public void updateFilterResults(
                     filterBindingSet.addBinding(binding);
                 }
             }
-            final String filterBindingSetString = converter.convert(filterBindingSet, filterVarOrder);
 
-            final Bytes row = encoder.encode( filterMetadata.getNodeId() + NODEID_BS_DELIM + filterBindingSetString );
+            final String filterBindingSetIdString = ID_CONVERTER.convert(filterBindingSet, filterVarOrder);
+            String filterBindingSetValueString = "";
+            if(childBindingSet instanceof VisibilityBindingSet) {
```

Comment:
We shouldn't do the type checking here. Assume it's always a VisibilityBindingSet once it's within the Fluo app. Lets update the BindingSet object that is passed into the updaters to require a VisibilityBindingSet.